pkg/jx/cmd/cmd_test.go

```diff
@@ -11,7 +12,9 @@ func TestNewJXCommand(t *testing.T) {
 	// TODO use mock stdio and expect some output ending in '>'
 	f := cmd.NewFactory()
 	f.SetOffline(true)
-	cmd := cmd.NewJXCommand(f, os.Stdin, os.Stdout, os.Stderr)
+	cmd := cmd.NewJXCommand(f, os.Stdin, os.Stdout, os.Stderr, []string{"prompt"})
+	// This hack allows us to override the command that will actually be executed.
+	// Normally it would come directly from os.Args
 	cmd.SetArgs([]string{"prompt"})
```

Comment:
So is the `SetArgs` call still needed?

Reply:
Yes. Otherwise it just tries to run the base command. I couldn't spot a better way to do it.